doc/topics/windows/windows-package-manager.rst

```diff
@@ -151,30 +151,100 @@ Maintaining Windows Repo Definitions in Git Repositories
-runner to compile the repository cache. Finally, use :py:func:`pkg.refresh_db
-<salt.modules.win_pkg.refresh_db>` on each minion to have them update their
-copy of the repository cache. Command examples are as follows:
+Package definitions are pulled down from the online repository by running the
+:mod:`winrepo.update_git_repos <salt.runners.winrepo.update_git_repos>` runner.
+This command is run on the master:
 
 .. code-block:: bash
 
     salt-run winrepo.update_git_repos
+
+This will pull down the software definition files for older minions
+(``salt-winrepo``) and new minions (``salt-winrepo-ng``). They are stored in the
+``file_roots`` under ``win/repo/salt-winrepo`` and
+``win/repo-ng/salt-winrepo-ng`` respectively.
+
+.. important::
+    If you have customized software definition files that aren't maintained in a
+    repository, those should be stored under ``win/repo`` for older minions and
+    ``win/repo-ng`` for newer minions. The reason for this is that the contents
+    of ``win/repo/salt-winrepo`` and ``win/repo-ng/salt-winrepo-ng`` are wiped
+    out every time you run a ``winrepo.update_git_repos``.
+
+    Additionally, when you run ``winrepo.genrepo`` and ``pkg.refresh_db`` the
+    entire contents under ``win/repo`` and ``win/repo-ng``, to include all
+    subdirectories, are used to create the message pack file.
+
+The next step (if you have older minions) is to create the message pack file for
+the repo (``winrepo.p``). This is done by running the
+:mod:`winrepo.genrepo <salt.runners.winrepo.genrepo>` runner. This is also run
+on the master:
+
+.. code-block:: bash
+
     salt-run winrepo.genrepo
-    salt winminion pkg.refresh_db
+
+.. note::
+    If you have only 2015.8.0 and newer minions, you no longer need to run
+    ``salt-run winrepo.genrepo`` on the master.
+
+Finally, you need to refresh the minion database by running the
+:py:func:`pkg.refresh_db <salt.modules.win_pkg.refresh_db>` command. This is run
+on the master as well:
+
+.. code-block:: bash
+
+    salt '*' pkg.refresh_db
+
+On older minions (older than 2015.8.0) this will copy the winrepo.p file down to
+the minion. On newer minions (2015.8.0 and newer) this will copy all the
+software definition files (.sls) down to the minion and then create the
+message pack file (``winrepo.p``) locally. The reason this is done locally is
+because the jinja needs to be parsed using the minion's grains.
+
+.. important::
+    Every time you modify the software definition files on the master, either by
+    running ``salt-run winrepo.update_git_repos``, modifying existing files, or
+    by creating your own, you need to refresh the database on your minions. For
+    older minions, that means running ``salt-run winrepo.genrepo`` and then
+    ``salt '*' pkg.refresh_db``. For newer minions (2015.8.0 and newer) it is
+    just ``salt '*' pkg.refresh_db``.
+
+.. note::
+If the ``winrepo.genrepo`` or the ``pkg.refresh_db`` fails, it is likely a
```

Comment:
I think you need some space here to match the other lines, otherwise I don't think this will render correctly.